File: src/main/java/org/hwabeag/hwaskyblock/schedules/HwaSkyBlockTask.java
package org.hwabeag.hwaskyblock.schedules;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.World;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitRunnable;
import org.hwabeag.hwaskyblock.HwaSkyBlock;
import org.hwabeag.hwaskyblock.config.ConfigManager;

import java.util.Objects;

public class HwaSkyBlockTask implements Runnable {

    FileConfiguration Config = ConfigManager.getConfig("setting");
    FileConfiguration SkyBlockConfig = ConfigManager.getConfig("skyblock");
    FileConfiguration PlayerConfig = ConfigManager.getConfig("player");
    String Prefix = ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(Config.getString("hwaskyblock-system.prefix")));

    @Override
    public void run() {
        for (Player player : Bukkit.getOnlinePlayers()) {
            String name = player.getName();
            if (PlayerConfig.get(name + ".skyblock") != null) {
                World world = player.getWorld();
                String world_name = world.getWorldFolder().getName();
                String[] number = world_name.split("\\.");
                if (Objects.equals(number[0], "HwaSkyBlock")) {
                    String block_to_id = number[1];
                    String weather = SkyBlockConfig.getString(block_to_id + ".setting.weather");
                    String time = SkyBlockConfig.getString(block_to_id + ".setting.time");
                    new BukkitRunnable() {
                        @Override
                        public void run() {
                            if (Objects.equals(weather, "clear")) {
                                world.setClearWeatherDuration(Integer.MAX_VALUE);
                            } else {
                                world.setClearWeatherDuration(0);
                            }
                            if (Objects.equals(weather, "rainy")) {
                                world.setStorm(true);
                                world.setWeatherDuration(Integer.MAX_VALUE);
                            } else {
                                world.setStorm(false);
                                world.setWeatherDuration(0);
                            }
                            if (Objects.equals(weather, "thunder")) {
                                world.setThunderDuration(Integer.MAX_VALUE);
                                world.setStorm(true);
                            } else {
                                world.setThunderDuration(0);
                            }
                            world.setThunderDuration(0);
                            world.setWeatherDuration(0);

                            if (Objects.equals(time, "morn")) {
                                long time_l = 1000;
                                world.setTime(time_l);
                            }
                            if (Objects.equals(time, "noon")) {
                                long time_l = 6000;
                                world.setTime(time_l);
                            }
                            if (Objects.equals(time, "evening")) {
                                long time_l = 18000;
                                world.setTime(time_l);
                            }
                        }
                    }.runTask(HwaSkyBlock.getPlugin());
                    String player_chunk = PlayerConfig.getString(name + ".skyblock.pos");
                    if (!Objects.equals(player_chunk, block_to_id)) {
                        if (SkyBlockConfig.get(block_to_id + ".leader") != null) {
                            String welcome_message = SkyBlockConfig.getString(block_to_id + ".welcome_message");
                            player.sendMessage(Prefix + " " + ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(welcome_message)));
                            String chunk_master = SkyBlockConfig.getString(block_to_id + ".leader");
                            player.sendTitle(Prefix, ChatColor.translateAlternateColorCodes('&', "&r주인장 &f: &e" + chunk_master));
                            PlayerConfig.set(name + ".skyblock.pos", block_to_id);
                            ConfigManager.saveConfigs();
                        }
                    }
                    PlayerConfig.set(name + ".skyblock.pos", block_to_id);
                    ConfigManager.saveConfigs();
                } else {
                    PlayerConfig.set(name + ".skyblock.pos", world_name);
                    ConfigManager.saveConfigs();
                }
            }
        }
    }
}

File: src/main/kotlin/schedules/HwaSkyBlockTask.kt
package org.hwabeag.hwaskyblock.schedules

import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.scheduler.BukkitRunnable
import org.hwabeag.hwaskyblock.HwaSkyBlock
import org.hwabeag.hwaskyblock.database.DatabaseManager
import org.hwabeag.hwaskyblock.database.config.ConfigManager
import java.util.*

class HwaSkyBlockTask : Runnable {
    var Config: FileConfiguration = ConfigManager.getConfig("setting")!!
    var Prefix: String = ChatColor.translateAlternateColorCodes(
        '&',
        Objects.requireNonNull<String?>(Config.getString("hwaskyblock-system.prefix"))
    )

    @Suppress("DEPRECATION")
    override fun run() {
        for (player in Bukkit.getOnlinePlayers()) {
            val name = player.name
            val skyblockId = DatabaseManager.getUserData("$name.skyblock", player, null)
            if (skyblockId != null) {
                val world = player.world
                val world_name = world.worldFolder.getName()
                val number: Array<String?> =
                    world_name.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                if (number[0] == "HwaSkyBlock") {
                    val block_to_id = number[1]
                    val weather = DatabaseManager.getSkyBlockData(block_to_id.toString(), "$block_to_id.setting.weather", "getSkyBlockWeather") as? String
                    val time = DatabaseManager.getSkyBlockData(block_to_id.toString(), "$block_to_id.setting.time", "getSkyBlockTime") as? String
                    object : BukkitRunnable() {
                        override fun run() {
                            if (weather == "clear") {
                                world.clearWeatherDuration = Int.Companion.MAX_VALUE
                            } else {
                                world.clearWeatherDuration = 0
                            }
                            if (weather == "rainy") {
                                world.setStorm(true)
                                world.weatherDuration = Int.Companion.MAX_VALUE
                            } else {
                                world.setStorm(false)
                                world.weatherDuration = 0
                            }
                            if (weather == "thunder") {
                                world.thunderDuration = Int.Companion.MAX_VALUE
                                world.setStorm(true)
                            } else {
                                world.thunderDuration = 0
                            }
                            world.thunderDuration = 0
                            world.weatherDuration = 0

                            if (time == "morn") {
                                val time_l: Long = 1000
                                world.time = time_l
                            }
                            if (time == "noon") {
                                val time_l: Long = 6000
                                world.time = time_l
                            }
                            if (time == "evening") {
                                val time_l: Long = 18000
                                world.time = time_l
                            }
                        }
                    }.runTask(HwaSkyBlock.plugin)
                    val player_chunk = DatabaseManager.getUserData("$name.skyblock.pos", player, "getPlayerPos") as? String
                    if (player_chunk != block_to_id) {
                        if (DatabaseManager.getSkyBlockData(block_to_id.toString(), "$block_to_id.leader", "getSkyBlockLeader") != null) {
                            val welcome_message = DatabaseManager.getSkyBlockData(block_to_id.toString(), "$block_to_id.welcome_message", "getSkyBlockWelcomeMessage") as? String
                            player.sendMessage(
                                "$Prefix " + ChatColor.translateAlternateColorCodes(
                                    '&',
                                    Objects.requireNonNull<String?>(welcome_message)
                                )
                            )
                            val chunk_master = DatabaseManager.getSkyBlockData(block_to_id.toString(), "$block_to_id.leader", "getSkyBlockLeader") as? String

                            player.sendTitle(
                                Prefix,
                                ChatColor.translateAlternateColorCodes('&', "&r주인장 &f: &e$chunk_master")
                            )
                            DatabaseManager.setUserData("$name.skyblock.pos", player, block_to_id.toString(), "setPlayerPos")
                        }
                    }
                    DatabaseManager.setUserData("$name.skyblock.pos", player, block_to_id.toString(), "setPlayerPos")
                } else {
                    DatabaseManager.setUserData("$name.skyblock.pos", player, world_name, "setPlayerPos")
                }
            }
        }
    }
}